src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -0,0 +1,410 @@
+          // Given that Docker only works under linux with 64bit,
+          // let's (for now) set these values to reflect that.
+          val labels = mesos.Labels.newBuilder
+            .addAllLabels(
+              Iterable(
+              mesos.Label.newBuilder.setKey("os").setValue("linux").build,
+              mesos.Label.newBuilder.setKey("arch").setValue("amd64").build
+            ).asJava)
+
+          val appc = mesos.Image.Appc.newBuilder.setName(im.id).setLabels(labels)
+
+          image.setType(mesos.Image.Type.APPC).setAppc(appc)
+      }
+
+      val mesosInfo = mesos.ContainerInfo.MesosInfo.newBuilder.setImage(image)
+      containerInfo.setMesos(mesosInfo)
+    }
+
+    containerInfo
+  }
+
+  private[this] def computeHealthCheck(
+    healthCheck: raml.HealthCheck,
+    endpoints: Seq[raml.Endpoint]): mesos.HealthCheck.Builder = {
+    val builder = mesos.HealthCheck.newBuilder
+
+    if (healthCheck.command.isDefined) {
+      builder.setType(mesos.HealthCheck.Type.COMMAND)
+
+      val command = mesos.CommandInfo.newBuilder
+
+      healthCheck.command.get.command match {
+        case raml.ShellCommand(shell) =>
+          command.setShell(true)
+          command.setValue(shell)
+        case raml.ArgvCommand(argv) =>
+          command.setShell(false)
+          if (argv.nonEmpty) {
+            command.setValue(argv.head)
+          }
+
+          if (argv.size > 1) {
+            command.addAllArguments(argv.tail.asJava)
+          }
+      }
+
+      builder.setCommand(command)
+    } else if (healthCheck.http.isDefined) {
+      builder.setType(mesos.HealthCheck.Type.HTTP)
+
+      val http = mesos.HealthCheck.HTTPCheckInfo.newBuilder
+
+      healthCheck.http.get.scheme.foreach(scheme => http.setScheme(scheme.value))
+      healthCheck.http.get.path.foreach(http.setPath)
+
+      builder.setHttp(http)
+    } else if (healthCheck.tcp.isDefined) {
+      builder.setType(mesos.HealthCheck.Type.TCP)
+
+      val tcp = mesos.HealthCheck.TCPCheckInfo.newBuilder
+
+      endpoints.find(_.name == healthCheck.tcp.get.endpoint).foreach{ endpoint =>
+        // TODO: determine if not in "HOST" mode and use the container port instead
+        endpoint.hostPort.foreach(tcp.setPort)
+      }
+
+      builder.setTcp(tcp)
+    }
+
+    builder
+  }
+
+  private[this] def taskContextEnv(
+    container: raml.MesosContainer,
+    version: Timestamp,
+    taskId: Instance.Id): Map[String, String] = {
+    Map(
+      "MESOS_TASK_ID" -> Some(taskId.idString),
+      "MARATHON_APP_ID" -> Some(container.name),
+      "MARATHON_APP_VERSION" -> Some(version.toString),
+      "MARATHON_APP_RESOURCE_CPUS" -> Some(container.resources.cpus.toString),
+      "MARATHON_APP_RESOURCE_MEM" -> Some(container.resources.mem.toString),
+      "MARATHON_APP_RESOURCE_DISK" -> Some(container.resources.disk.toString),
+      "MARATHON_APP_RESOURCE_GPUS" -> Some(container.resources.gpus.toString)
```

Comment:
I think there's a dc/os EE dependency on MARATHON_APP_ID; let's keep this
in mind in case we rename

On Mon, Sep 12, 2016 at 8:16 AM, Matthias Veit <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala
> <https://github.com/mesosphere/marathon/pull/4353#discussion_r78360843>:
>
> > +
> > +    builder
> > +  }
> > +
> > +  private[this] def taskContextEnv(
> > +    container: raml.MesosContainer,
> > +    version: Timestamp,
> > +    taskId: Instance.Id): Map[String, String] = {
> > +    Map(
> > +      "MESOS_TASK_ID" -> Some(taskId.idString),
> > +      "MARATHON_APP_ID" -> Some(container.name),
> > +      "MARATHON_APP_VERSION" -> Some(version.toString),
> > +      "MARATHON_APP_RESOURCE_CPUS" -> Some(container.resources.cpus.toString),
> > +      "MARATHON_APP_RESOURCE_MEM" -> Some(container.resources.mem.toString),
> > +      "MARATHON_APP_RESOURCE_DISK" -> Some(container.resources.disk.toString),
> > +      "MARATHON_APP_RESOURCE_GPUS" -> Some(container.resources.gpus.toString)
>
> We probably should rename MARATHON_APP_ to MARATHON_CONTAINER_ ?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4353/files/0a69499a62feb921a20eff42c484a997f004cd01#r78360843>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLFAQGcQ3YGqWEGKMnD-1DJUtEu0Mks5qpUKJgaJpZM4J6bJ5>
> .
>